File: ApplicationAlgorithm/test/unit1/unit1.cpp
#include <iostream>

using namespace std;

int fibonacci(int n)
{
	if (n < 2)
		return n;
	return fibonacci(n - 1) + fibonacci(n - 2);
}

int fuction(int k)
{
	if (k < 0)
		return 0;
	int i = 0;
	while (true)
	{
		if (fibonacci(i) > k)
			return fibonacci(i);
		i++;
	}
}

int main()
{
	int k;
	cout << "Nhap k: ";
	cin >> k;
	cout << "Ket qua: " << fuction(k);
	return 0;
}

File: ApplicationAlgorithm/test/unit1/unit1.py
def fibonacci(n):
    if n < 2:
        return n
    else:
        return fibonacci(n - 1) + fibonacci(n - 2)

def function(k):
    if k < 1:
        return 0
    else:
        i = 0
        while (fibonacci(i) <= k):
            i += 1
        return fibonacci(i)

k = int(input())
print(function(k))